### ISSUE TITLE
Unexpected end of JSON input when trying to initialise the account

### ISSUE THREAD

**Issue by wizy69** [586837r/node-red-contrib-alexa-remote2#187]
on a clean install, using proxy authentication. After I open the requested browser session to generate the cookie( message confirmed) . Account therefore does not initialise 

Syntax error on screen, debug:
node: 146d4052.3ded4
msg : string[28]
"Unexpected end of JSON input"

Have tried other authentiction (cookies and proxy) same result. Even after node red restart.  Node red latest on Home Assist Virtual Env
AU Country utilised. Even tried others (interesting that authentication still produce positive cookie message?

**Comment by mainres07** [comment]
Same problem here.

**Comment by bdydrp** [comment]
Same here for an AUS account..
Same error using the init flow

EDIT: definitely a aussie account issue!! When i use my USA alexa details it works!!

**Comment by wizy69** [comment]
I managed to get it to work. Did a new  machine and started from scratch install of HA as there was an issue with Supervisor Logs ( not sure if related). Got it working by using cookies method, And guidelines explained https://www.youtube.com/watch?v=Yg7dv4OOFAY&t=742s, 12:40 onwards on getting cookie.  Also, don't inject right away. Once the routine account settings are updated, wait for it ti display "Ready" under the routine. Then it should have initialized and device can be selected.
So to summarise, have timestamp inject and Alexa Routine Mode Nodes. On the routine node, i have auth method: cookie selected; cookie text copied from and including session-id= text; auto :on , events:on, service:alexa.amazon.com.au; page: amazon.com.au
Will need to see how long before cookie gets refresehd, and will check other examples where cookies autorefreshed

**Comment by bdydrp** [comment]
Manual cookie method will work, but it wont auto update. 
After 4-5 days you will need to manually refresh the cookie

**Comment by wizy69** [comment]
Even Manual cookie failed after a while ( suspect account password change), even if new cookie manually sourced through another browser session. Have reverted to email/password authentication ( no 2fa) and working so-far.  I suspect there is no cokkie being passed or generated via the nodered proxy authentication method, as logs show error: Unexpected token s in JSON at position 0.
If the cokkie session is being passed back is blank , or jsonwrite command is empty it will cause a JSON failed error. Any luck getting this to work for others using 2fa?

**Comment by bdydrp** [comment]
If use my American Amazon account, it generates a cookie, so I know my init
flow works.
I dont know how the 2 account cookies are different, but there must be
something

**Comment by 877dev** [comment]
I'm getting the same with UK account, and also this message sometimes:

![image](https://user-images.githubusercontent.com/46575282/105724932-48ce3d80-5f20-11eb-9602-982bee2a642a.png)

**Comment by GurjBahia** [comment]
Same here, UK account.  It started happening multiple times a day during January.
I have been using an hourly Refresh and Re-Init (copied from someone else here a while ago).
![image](https://user-images.githubusercontent.com/12628955/105824990-f7bd5880-5fb6-11eb-8c50-0fdbc0ba4413.png)
Interestingly "Restart Flows" *sometimes* succeeds in login, but not always.
Not sure why restarting flows should work.

**Comment by mcarame** [comment]
Same here, IT Account.

> Same here, UK account. It started happening multiple times a day during January.
> I have been using an hourly Refresh and Re-Init (copied from someone else here a while ago).
> ![image](https://user-images.githubusercontent.com/12628955/105824990-f7bd5880-5fb6-11eb-8c50-0fdbc0ba4413.png)
> Interestingly "Restart Flows" _sometimes_ succeeds in login, but not always.
> Not sure why restarting flows should work.


Hi, @GurjBahia could you please share here the code for the Refresh and the Re-Init?
Thanks!

**Comment by GurjBahia** [comment]
> Same here, IT Account.
> 
> > Same here, UK account. It started happening multiple times a day during January.
> > I have been using an hourly Refresh and Re-Init (copied from someone else here a while ago).
> > ![image](https://user-images.githubusercontent.com/12628955/105824990-f7bd5880-5fb6-11eb-8c50-0fdbc0ba4413.png)
> > Interestingly "Restart Flows" _sometimes_ succeeds in login, but not always.
> > Not sure why restarting flows should work.
> 
> Hi, @GurjBahia could you please share here the code for the Refresh and the Re-Init?
> Thanks!

`[{"id":"e1fe367f.c0c468","type":"alexa-remote-init","z":"8dfc23dd.12d368","name":"","account":"","option":"refresh","x":640,"y":1320,"wires":[["6b036a0e.0ec07c"]]},{"id":"95e9d8e0.ec142","type":"inject","z":"8dfc23dd.12d368","name":"1hr Refresh","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"3600","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":410,"y":1320,"wires":[["e1fe367f.c0c468"]]},{"id":"f35507e7.5b5258","type":"catch","z":"8dfc23dd.12d368","name":"Catch Refresh Errors","scope":["95e9d8e0.ec142"],"uncaught":false,"x":470,"y":1400,"wires":[["f03ef29e.ecf8b"]]},{"id":"f03ef29e.ecf8b","type":"alexa-remote-init","z":"8dfc23dd.12d368","name":"","account":"","option":"fresh","x":720,"y":1400,"wires":[["2aa8eaf1.8eb296"]]},{"id":"2aa8eaf1.8eb296","type":"debug","z":"8dfc23dd.12d368","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","x":990,"y":1400,"wires":[]},{"id":"6b036a0e.0ec07c","type":"debug","z":"8dfc23dd.12d368","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","x":990,"y":1320,"wires":[]},{"id":"988670bd.17836","type":"inject","z":"8dfc23dd.12d368","name":"Manual","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":430,"y":1480,"wires":[["f03ef29e.ecf8b"]]}]`

Corrected title of Manual
![image](https://user-images.githubusercontent.com/12628955/105904610-43a0e980-6019-11eb-8206-5f8b4a0052bf.png)

**Comment by mcarame** [comment]
> > Same here, IT Account.
> > > Same here, UK account. It started happening multiple times a day during January.
> > > I have been using an hourly Refresh and Re-Init (copied from someone else here a while ago).
> > > ![image](https://user-images.githubusercontent.com/12628955/105824990-f7bd5880-5fb6-11eb-8c50-0fdbc0ba4413.png)
> > > Interestingly "Restart Flows" _sometimes_ succeeds in login, but not always.
> > > Not sure why restarting flows should work.
> > 
> > 
> > Hi, @GurjBahia could you please share here the code for the Refresh and the Re-Init?
> > Thanks!
> 
> `[{"id":"e1fe367f.c0c468","type":"alexa-remote-init","z":"8dfc23dd.12d368","name":"","account":"","option":"refresh","x":640,"y":1320,"wires":[["6b036a0e.0ec07c"]]},{"id":"95e9d8e0.ec142","type":"inject","z":"8dfc23dd.12d368","name":"1hr Refresh","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"3600","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":410,"y":1320,"wires":[["e1fe367f.c0c468"]]},{"id":"f35507e7.5b5258","type":"catch","z":"8dfc23dd.12d368","name":"Catch Refresh Errors","scope":["95e9d8e0.ec142"],"uncaught":false,"x":470,"y":1400,"wires":[["f03ef29e.ecf8b"]]},{"id":"f03ef29e.ecf8b","type":"alexa-remote-init","z":"8dfc23dd.12d368","name":"","account":"","option":"fresh","x":720,"y":1400,"wires":[["2aa8eaf1.8eb296"]]},{"id":"2aa8eaf1.8eb296","type":"debug","z":"8dfc23dd.12d368","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","x":990,"y":1400,"wires":[]},{"id":"6b036a0e.0ec07c","type":"debug","z":"8dfc23dd.12d368","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"payload","x":990,"y":1320,"wires":[]},{"id":"988670bd.17836","type":"inject","z":"8dfc23dd.12d368","name":"Manual","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":430,"y":1480,"wires":[["f03ef29e.ecf8b"]]}]`
> 
> Corrected title of Manual
> ![image](https://user-images.githubusercontent.com/12628955/105904610-43a0e980-6019-11eb-8206-5f8b4a0052bf.png)

Thank you very much!

**Comment by elchanly** [comment]
Try this flow:  this flow checks if account is not initialized and rejoins automagically.
https://github.com/driagi/node-red-contrib-alexa-remote2

**Comment by wizy69** [comment]
Anyone know how to debug to try see what is causing the issue, or which file is used in initiating the routine so that I can debug further. I want to return the cokkie txt or see where it is converting to JSON Input to see if its a character limit or perhaps it does not need to e converted

**Comment by flashtoday** [comment]
> Anyone know how to debug to try see what is causing the issue, or which file is used in initiating the routine so that I can debug further. I want to return the cokkie txt or see where it is converting to JSON Input to see if its a character limit or perhaps it does not need to e converted

I know this is old, but did you ever have any luck debugging this?

**Comment by abphd** [comment]
Same for me AU. Any solutions?

**Comment by bdydrp** [comment]
> Same for me AU. Any solutions?

Update to the applestrudel fork!
This works with au accounts. Mine has been working flawless for 3 weeks now

**Comment by wizy69** [comment]
Issue went away not sure if latest version resolved or host software update resolved issue


Sent from Yahoo Mail for iPhone


On Tuesday, August 23, 2022, 13:30, bdydrp ***@***.***> wrote:





Same for me AU. Any solutions?


Update to the applestrudel fork!
This works with au accounts. Mine has been working flawless for 3 weeks now
